server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java

```diff
@@ -152,6 +154,20 @@ public ActionRequestValidationException validate() {
         if (featureStates == null) {
             validationException = addValidationError("featureStates is missing", validationException);
         }
+        if (renameReplacement != null) {
+            int byteLength = renameReplacement.getBytes(StandardCharsets.UTF_8).length;
+
+            if (renameReplacement.getBytes(StandardCharsets.UTF_8).length > MAX_INDEX_NAME_BYTES) {
+                validationException = addValidationError(
+                    "rename_replacement UTF-8 byte length ["
+                        + byteLength
+                        + "] exceeds maximum allowed length ["
+                        + MAX_INDEX_NAME_BYTES
+                        + "] bytes",
+                    validationException
+                );
+            }
```

Comment:
Due to usage of back-references and escapes, the result index name can be shorter than 255 when `renameReplacement` is longer than 255. So we could be preventing some legal index names in edge cases. I think it might be OK though since such long index names are very rare.

Reply:
Yeah. We could do something along that line. I'd prefer to be a bit more strict, e.g. allowing an extra buffer of 10 chars.